File: problems/1700_number_of_students_unable_to_eat_lunch.py
class Solution:
    def countStudents(self, students: list[int], sandwhiches: list[int]) -> int:
        n = len(students)
        studentCounts: dict[int, int] = {0: 0, 1: 0}
        for student in students:
            studentCounts[student] += 1
        for i in range(n):
            if studentCounts[sandwhiches[i]] == 0:
                return n - i
            studentCounts[sandwhiches[i]] -= 1
        return 0

File: problems/1700-number-of-students-unable-to-eat-lunch.ts
function countStudents(students: number[], sandwiches: number[]): number {
  const n = students.length
  const studentCounts: [number, number] = [0, 0]

  for (const student of students) {
    studentCounts[student] += 1
  }

  for (let i = 0; i < n; i++) {
    const sandwich = sandwiches[i]
    if (studentCounts[sandwich] === 0) {
      return n - i
    }
    studentCounts[sandwich] -= 1
  }

  return 0
}
